server/src/main/java/org/opensearch/index/mapper/NumberFieldMapper.java

```diff
@@ -212,7 +221,7 @@ public Query termsQuery(String field, List<Object> values) {
                 for (int i = 0; i < values.size(); ++i) {
                     v[i] = parse(values.get(i), false);
                 }
-                return HalfFloatPoint.newSetQuery(field, v);
+                return FloatField.newSetQuery(field, v);
```

Comment:
I believe this is not right, `FloatField` would use `FloatPoint` but we need `HalfFloatPoint`